Sources/Features/Client/BacktraceOomWatcher.swift

```diff
@@ -0,0 +1,220 @@
+import Foundation
+
+final class BacktraceOomWatcher {
+    
+    private var state: ApplicationInfo
+    
+    let lowMemoryFilePrefix = "_lowMemory"
+    private(set) static var oomFilePath: URL? = getStatusFilePath()
+    
+    private(set) var crashReporter: CrashReporting
+    private(set) var attributesProvider: AttributesProvider
+    private(set) var backtraceApi: BacktraceApi
+    private let repository: PersistentRepository<BacktraceReport>
+    
+    init(
+        repository: PersistentRepository<BacktraceReport>,
+        crashReporter: CrashReporting,
+        attributes: AttributesProvider,
+        backtraceApi: BacktraceApi) {
+        self.crashReporter = crashReporter
+        self.attributesProvider = attributes
+        self.backtraceApi = backtraceApi
+        self.repository = repository
+        
+        // set default state
+        state = ApplicationInfo()
+        
+        // set status file url
+        if BacktraceOomWatcher.oomFilePath == nil {
+            // database path will point out sqlite database path
+            // oom status should exist in the same directory where database exists.
+            BacktraceOomWatcher.oomFilePath = self.repository.url.deletingLastPathComponent().absoluteURL
+                .appendingPathComponent("BacktraceOOMState.plist")
+        }
+    }
+    
+    internal static func clean() {
+        if let oomFilePath = BacktraceOomWatcher.oomFilePath {
+            // ignore errors or use do/catch block to handle errors more gracefully
+            try? FileManager.default.removeItem(at: oomFilePath)
+        }
+    }
+    
+    internal static func getAppVersion() -> String {
+        if let appVersion = Bundle.main.object(forInfoDictionaryKey: kCFBundleVersionKey as String) as? String {
+            return appVersion
+        } else {
+            return ""
+        }
+    }
+    
+    public func start() {
+        sendPendingOomReports()
+        // override previous application state after reading all information
+        saveState()
+    }
+    
+    internal static func getStatusFilePath() -> URL? {
+        // oom status file is available in application cache dir - the same dir
+        // where we store attributes
+        guard let cacheDir = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        return cacheDir.appendingPathComponent("BacktraceOomState.plist")
+    }
+    
+    private func sendPendingOomReports() {
+        
+        // if oom state file doesn't exist it means that we deleted it to
+        // prevent sending false oom crashes
+        if let oomFilePath = BacktraceOomWatcher.oomFilePath {
+            if !FileManager.default.fileExists(atPath: oomFilePath.path) {
```

Comment:
It can be a one-liner:

```swift
if let oomFilePath = BacktraceOomWatcher.oomFilePath, !FileManager.default.fileExists(atPath: oomFilePath.path) {
```